core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -1028,9 +1037,25 @@ private RexNode simplifyIs(RexCall call, RexUnknownAs unknownAs) {
       switch (a.getKind()) {
       case LITERAL:
         return rexBuilder.makeLiteral(!((RexLiteral) a).isNull());
+      case DIVIDE: {
```

Comment:
note that floating point division behaves like the STRONG policy, since division by 0 returns an infinity or NaN.